### PR TITLE
Retry mechanism in ssh service for script_output

### PR DESCRIPTION
**Pull request by hadeskun**
Retried on failure in the script_output for ssh
Ticket: https://jira.suse.com/browse/TEAM-9651
- Verification run: https://openqaworker15.qa.suse.cz/tests/299921#

### CODE REVIEW

File: lib/publiccloud/instance.pm [hunk 1]
```diff
@@ -848,5 +848,25 @@ sub do_systemd_analyze_time {
     return @ret;
 }
 
+sub retry_on_script_output {
```
Comment (ricardobranco777) [inline lib/publiccloud/instance.pm:851]: I'm afraid this is a very generic name for a function that runs only a hard-coded command. =/

File: lib/publiccloud/instance.pm [hunk 2]
```diff
@@ -848,5 +848,25 @@ sub do_systemd_analyze_time {
...
+    my ($self, %args) = @_;
+    my $retries = 10;
+    my $delay_command = 20;
```
Comment (ricardobranco777) [inline lib/publiccloud/instance.pm:854]: Is this in seconds?  Then I think 20 is a bit too much.  Why not try with a lower value using exponential backoff?

**Review by asmorodskyi — CHANGES_REQUESTED** [review]
I am afraid we getting into unexpected recursion here . we ALREADY inside loop which will **retry** this command over and over again until timeout finishes so what you are doing here is already done in line 425 :man_shrugging:

File: lib/publiccloud/instance.pm (hunk 1 above)
Comment (m-dati) [reply lib/publiccloud/instance.pm:851]: Sorry, but this coommand was not intended here to have a retry: the failure is managed into the while loop!
Why do you need this retry now?

Comment (m-dati) [reply lib/publiccloud/instance.pm:851]: I cannot read the SD ticket: why do you think this change is needed?

File: lib/publiccloud/instance.pm [hunk 3]
```diff
@@ -848,5 +848,25 @@ sub do_systemd_analyze_time {
...
+    my $sys_response = "";
+
+    while ($retries--) {
+        $sys_response = $self->ssh_script_output(
+            cmd => 'sudo systemctl is-system-running',
```
Comment (m-dati) [inline lib/publiccloud/instance.pm:859]: Uneeded: it is already done in the original code loop from L425-849, long studied, tested, checked, reviewed and correctly running until now.

**Review by m-dati — CHANGES_REQUESTED** [review]
Uneeded: it is already done in the original code loop from L425-849, long studied, tested, checked, reviewed and correctly running until now.

**Pull request closed without merging**